### ISSUE TITLE
Do you have object detection bounding box visualization results?

### ISSUE THREAD

**Issue by c0nn3r** [andreaazzini/retinanet.pytorch#2]
Trying to debug my implementation's training on COCO, would love to see if you got this repo working on VOC.

**Comment by andreaazzini** [comment]
So, I'm still far from achieving good results on VOC (I had the time to complete only 72 epochs). Some examples (good, promising and bad).

![2008_001271](https://user-images.githubusercontent.com/8004379/30342425-b7e783da-97ae-11e7-8685-ea5c7d722a62.jpg)
![2012_000745](https://user-images.githubusercontent.com/8004379/30342543-152ef0d2-97af-11e7-8ec3-36a7447dfd95.jpg)
![2012_002009](https://user-images.githubusercontent.com/8004379/30342545-178c42d0-97af-11e7-97e8-b3dca7fe4c3e.jpg)

My gut feeling is that something is still wrong in the encoder.

**Comment by andreaazzini** [comment]
@c0nn3r any better results on your side?

**Comment by c0nn3r** [comment]
Nothing yet, having trouble batching the bounding boxes. Does it only work when only one bounding box is needed to detect (or a lower number)? I'm thinking of writing tests for `encoder.py` and just brute force a working solution. I'm going to switch to getting VOC working.

**Comment by c0nn3r** [comment]
Also @andreaazzini, what are you using to produce the images shown?

**Comment by andreaazzini** [comment]
@c0nn3r https://github.com/andreaazzini/retinanet.pytorch/blob/328f63e12fdcd9e5b11f8828828b302245b13cf8/demo.py
I removed `demo.py` from master because I'm working on an evaluation script, and the demo script needs to be changed accordingly.

**Comment by c0nn3r** [comment]
@andreaazzini I took a look at the loss function and found some problems:
- I don't think we are normalizing focal loss over the number of background detections / ground-truth boxes (it was italicised in the paper, so it must be important!).
- The paper claims to have ~100K anchor predictions for the box subnet, that doesn't seem to match our ~47961.

**Comment by andreaazzini** [comment]
@c0nn3r Are you using my encoder or @kuangliu's?

**Comment by c0nn3r** [comment]
@andreaazzini I'm using your encoder
